src/main/java/ch/njol/skript/classes/data/DefaultFunctions.java

```diff
@@ -711,6 +705,227 @@ public String[] executeSimple(Object[][] params) {
+				if (number < -170) {
+					return new Number[]{Double.NEGATIVE_INFINITY};
+				} else if (number < 0) {
+					negative = true;
+					number = Math.abs(number);
+				} else if (number <= 1) { // 0 and 1
+					return new Number[]{1};
+				} else if (number > 170) {
+					return new Number[]{Double.POSITIVE_INFINITY};
+				}
+				Double result = 1d;
+				for (double i = number; i > 1; i--) {
+					if (result.isInfinite() || result.isNaN())
+						break;
+					result *= i;
+				}
+				if (negative)
+					result *= -1d;
+				return new Number[]{result};
+			}
+		})
+			.description(
+				"Get the factorial of a number. ",
+				"Getting the factorial of any number above 20 goes into the Double range leading to floating points.",
+				"Any number after 170 will always return Infinity.",
+				"Should not be used to calculate permutations or combinations manually."
+			)
+			.examples(
+				"factorial(0) = 1",
+				"factorial(3) = 3*2*1 = 6",
+				"factorial(5) = 5*4*3*2*1 = 120",
+				"factorial(171) = Infinity"
+			)
+			.since("INSERT VERSION");
+
+		Functions.registerFunction(new SimpleJavaFunction<Number>("root", new Parameter[]{
+			new Parameter<>("n", DefaultClasses.NUMBER, true, null),
+			new Parameter<>("number", DefaultClasses.NUMBER, true, null)
+		}, DefaultClasses.NUMBER, true) {
+			@Override
+			public Number @Nullable [] executeSimple(Object[][] params) {
+				Double n = ((Number) params[0][0]).doubleValue();
+				Double number = ((Number) params[1][0]).doubleValue();
+				if (n == 0) {
+					return null;
+				} else if (n == 1) {
+					return new Number[]{number};
+				} else if (n == 2) {
+					return new Number[]{Math.sqrt(number)};
+				}
+				return new Number[]{Math.pow(number, (1 / n))};
+			}
+		})
+			.description("Calculates the <i>n</i>th root of a number.")
+			.examples(
+				"root(2, 4) = 2 # same as sqrt(4)",
+				"root(4, 16) = 2",
+				"root(-4, 16) = 0.5 # same as 16^(-1/4)"
+			)
+			.since("INSERT VERSION");
+
+		Functions.registerFunction(new SimpleJavaFunction<Number>("permutation", new Parameter[]{
+			new Parameter<>("options", DefaultClasses.NUMBER, true, null),
+			new Parameter<>("size", DefaultClasses.NUMBER, true, null)
+		}, DefaultClasses.NUMBER, true) {
+			@Override
+			public Number @Nullable [] executeSimple(Object[][] params) {
+				Double options = ((Number) params[0][0]).doubleValue();
+				Double size = ((Number) params[1][0]).doubleValue();
+				if (size > options || size < 0) {
+					return null;
+				} else if (size.equals(0d)) {
+					return new Number[]{1};
+				} else if (size.equals(1d)) {
+					return new Number[]{options};
+				}
+				Double result = 1d;
+				for (double i = options; i > options - size; i--) {
+					if (result.isInfinite() || result.isNaN())
+						break;
+					result *= i;
+				}
+				return new Number[]{result};
+			}
+		})
+			.description(
+				"Get the number of possible ordered arrangements from 1 to 'options' with each arrangement having a size equal to 'size'",
```

Comment:
I'd like a line explaining the diff between perm and comb, maybe something like `(3, 2) and (2, 3) are considered the same combination, but two different permutations` or something along that line.